src/control/ros/Conversions.cpp

```diff
@@ -311,6 +344,23 @@ std::unique_ptr<SplineTrajectory> toSplineJointTrajectory(
 
     rosJointToMetaSkeletonJoint.emplace(
         std::make_pair(trajIndex, metaSkeletonIndex));
+    specifiedMetaSkeletonJoints.insert(metaSkeletonIndex);
+  }
+
+  // Add unspecified joint mappings to rosJointToMetaSkeletonJoint
+  std::vector<size_t> unspecifiedMetaSkeletonJoints;
+  if (static_cast<std::size_t>(startPositions.size()) == numControlledJoints && numTrajectoryJoints < numControlledJoints)
```

Comment:
Nitpick: We can save this check by storing a flag whether we're in padding mode or not. The mode can be determined in the above error checking.